masu/database/report_db_accessor.py

```diff
@@ -0,0 +1,284 @@
+
+        Returns:
+            (dict): The data with values converted to required types
+
+        """
+        column_types = self.report_schema.column_types[table_name]
+
+        for key, value in data.items():
+            if not value:
+                data[key] = None
+                continue
+            if column_types.get(key) == int:
+                data[key] = self._convert_value(value, int)
+            elif column_types.get(key) == float:
+                data[key] = self._convert_value(value, float)
+            elif column_types.get(key) == Decimal:
+                data[key] = self._convert_value(value, Decimal)
+
+        return data
+
+    def _convert_value(self, value, column_type):
+        """Convert a single value to the specified column type.
+
+        Args:
+            value (var): A value of any type
+            column_type (type) A Python type
+
+        Returns:
+            (var): The variable converted to type or None if conversion fails.
+
+        """
+        try:
+            value = column_type(value)
+        except ValueError as err:
+            LOG.warning(err)
+            value = None
+        return value
+
+    def get_current_cost_entry_bill(self):
+        """Get the most recent cost entry bill object."""
+        billing_start = getattr(
+            self.report_schema.reporting_awscostentrybill,
+            'billing_period_start'
+        )
+        return self._get_db_obj_query('reporting_awscostentrybill')\
+            .order_by(billing_start.desc())\
+            .first()
+
+    def get_cost_entries(self):
+        """Make a mapping of cost entries by start time."""
+        interval_start = getattr(
+            self.report_schema.reporting_awscostentry,
+            'interval_start'
+        )
+        cost_entries = self._get_db_obj_query('reporting_awscostentry')\
+            .order_by(interval_start.desc())\
+            .all()
+
+        return {entry.interval_start.strftime(self._datetime_format): entry.id
+                for entry in cost_entries}
+
+    def get_products(self):
+        """Make a mapping of product sku to product objects."""
+        columns = ['id', 'sku']
+        products = self.session.query()
+        products = self._get_db_obj_query('reporting_awscostentryproduct',
+                                          columns=columns).all()
+
+        return {product.sku: product.id for product in products}
+
+    def get_pricing(self):
+        """Make a mapping of pricing values string to pricing objects."""
+        pricing = self._get_db_obj_query('reporting_awscostentrypricing').all()
+
+        return {f'{p.public_on_demand_cost}-{p.public_on_demand_rate}-{p.term}-{p.unit}': p.id
+                for p in pricing}
+
+    def get_reservations(self):
+        """Make a mapping of reservation ARN to reservation objects."""
+        columns = ['id', 'reservation_arn']
+        reservs = self._get_db_obj_query('reporting_awscostentryreservation',
```

Comment:
Down the road we should look at making these table names be variables instead of hard coded strings.  Might make managing changes easier.